common/s3/aws/aws_s3_client.go

```diff
@@ -23,38 +24,44 @@ const (
 )
 
 var (
-	once              sync.Once
-	ref               *client
-	ErrObjectNotFound = errors.New("object not found")
+	once sync.Once
+	ref  *awsS3Client
 )
 
-type Object struct {
-	Key  string
-	Size int64
-}
+// An implementation of s3common.S3Client using AWS S3.
+type awsS3Client struct {
+	logger logging.Logger
 
-type client struct {
-	cfg      *commonaws.ClientConfig
+	// Amazon's S3 client implementation.
 	s3Client *s3.Client
 
 	// concurrencyLimiter is a channel that limits the number of concurrent operations.
 	concurrencyLimiter chan struct{}
-
-	logger logging.Logger
 }
 
-var _ Client = (*client)(nil)
+var _ s3common.S3Client = (*awsS3Client)(nil)
+
+// NewAwsS3Client creates a new S3Client that talks to AWS S3.
+func NewAwsS3Client(
+	ctx context.Context,
+	logger logging.Logger,
+	endpointUrl string,
```

Comment:
Any specific reason you got rid of the `cfg commonaws.ClientConfig`, and passed everything in separately instead?